libshaderc/src/shaderc_private.h

```diff
@@ -42,6 +43,46 @@ struct shaderc_compilation_result {
       shaderc_compilation_status_null_result_object;
 };
 
+// Compilation result class using a vector for holding the compilation
+// output data.
+class shaderc_compilation_result_vector : public shaderc_compilation_result {
```

Comment:
Yeah, make sense for me. But I'd like to put that as a separate PR since this one is already fairly large.